### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by lenzloesche**
I wrote some tests for components and pages I already programmed last week or earlier.

### CODE REVIEW

File: components/StyledButton/StyledButton.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  expect(children).toBeInTheDocument();
+});
+
+test("on click a function is called", () => {
+  testFunction = jest.fn();
+  render(<StyledButton onClick={testFunction}>TestButton</StyledButton>);
+  const button = screen.getByText(/TestButton/i);
+  fireEvent.click(button);
+  expect(testFunction).toHaveBeenCalled();
+});
```
Comment (Satttoshi) [inline components/StyledButton/StyledButton.test.js:16]: Test fails. onClick function is missing, probably missing 
a
`const testfunction = jest.fn();`

**Review by Satttoshi — CHANGES_REQUESTED** [review]
Very nice test, one test fails for me though. Please check your test of your StyledButton component.

image

File: components/StyledButton/StyledButton.test.js (hunk 1 above)
Comment (lenzloesche) [reply components/StyledButton/StyledButton.test.js:16]: thanks for pointing that out. thats odd because I ran the tests and they worked.

**Review by actyralf — CHANGES_REQUESTED** [review]
Kudos for writing tests!
Please have a look at my comment regarding the use of `act()`.

File: __tests__/exerciseSearch.test.js [hunk 2]
```diff
@@ -13,3 +10,46 @@ test("see if search bar is rendered.", () => {
...
+test("if I search and find a result, there is a favorite button", async () => {
+  render(
+    <ExerciseSearch favoriteExercises={[]} setFavoriteExercises={() => {}} />
+  );
+  const searchButton = screen.getByRole("button", {name: /search/i});
+  const searchBar = screen.getByRole("textbox");
+  fireEvent.change(searchBar, {target: {value: "TestText"}});
+  await act(async () => {
+    await fireEvent.click(searchButton);
+  });
```
[the remark this answers is not in the document]
Comment (lenzloesche) [reply __tests__/exerciseSearch.test.js:49]: thanks, I will change it.

**Review by actyralf — APPROVED** [review]
Well done!

**Pull request merged**